docs/restrict-cross-namespace-associations.asciidoc

```diff
@@ -0,0 +1,97 @@
+
+[float]
+==== Enabling cross namespace association restriction
+
+This feature is disabled by default. To enable it, start the operator with the `--enforce-rbac-on-refs` flag.
+
+NOTE: This feature only provides an access control for resources deployed across two different namespaces. Associations can still be freely created between resources deployed in a same namespace.
+
+Once enabled, associations can be created as long as the `ServiceAccount` specified in the associated resources can execute a `get` request against the referenced Elasticsearch object.
+
+IMPORTANT: ECK unbinds any association that is not allowed. If you have some existing associations, do not enable this feature without creating the required `Roles` and `RoleBindings` as described in the following sections.
+
+[float]
+=== Specify a ServiceAccount in an associated resource
+
+First create a `ServiceAccount`:
+
+[source,sh]
+----
+> kubectl create serviceaccount associated-resource-sa
+----
+
+Then set the `serviceAccountName` field in the associated resource to specify which `ServiceAccount` is used to create the association:
+
+[source,yaml,subs="attributes"]
+----
+apiVersion: <kind>.k8s.elastic.co/{eck_crd_version}
+kind: <Kind>
+metadata:
+  name: associated-resource
+  namespace: associated-resource-ns
+spec:
+ ...
+  elasticsearchRef:
+    name: "elasticsearch-sample"
+    namespace: "elasticsearch-ns"
+  # Service account used by this resource to get access to an Elasticsearch cluster
+  serviceAccountName: associated-resource-sa
+----
+
+NOTE: You can find https://github.com/elastic/cloud-on-k8s/blob/master/config/recipes/associations-rbac/apm_es_kibana_rbac.yaml[a complete example in the ECK GitHub repository].
+
+NOTE: If the `serviceAccountName` is not set then ECK uses the `ServiceAccount` called `default`
+
+[float]
+==== Allowing associations to be created
+
+Create a `ClusterRole` to allow a `get` request to be run against some Elasticsearch objects:
+
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: elasticsearch-association
+rules:
+  - apiGroups:
+      - elasticsearch.k8s.elastic.co
+    resources:
+      - elasticsearches
+    verbs:
+      - get
+----
+
+Then, to allow the `ServiceAccount` of the associated resource to use this role create the following `RoleBinding`:
+
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: allow-associated-resource-from-remote-namespace
+  namespace: elasticsearch-ns
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: elasticsearch-association
+subjects:
+  - kind: ServiceAccount
+    name: associated-resource-sa
+    namespace: associated-resource-ns
+----
+
+The associated resource `associated-resource` is now allowed to create an association with any Elasticsearch cluster in the namespace `elasticsearch-ns`.
```

Comment:
I wonder if readers have at this point already forgotten what we mean by "associated resource" and whether it would be better to use a a concrete example instead (i.e. Kibana or APMServer) and mention that it applies to any association between resources we support?

Reply:
I added the following sentence right after the resource example:

> In the above example, `associated-resource` can be of any `Kind` that requires an association to be created (e.g. `Kibana` or the `APMServer`).

Let me know if you think it's enough